lib/mix/lib/mix/tasks/help.ex

```diff
@@ -194,13 +205,38 @@ defmodule Mix.Tasks.Help do
   end
 
   defp build_alias_doc_list(aliases) do
-    Enum.reduce(aliases, {[], 0}, fn {alias_name, _task_name}, {docs, max} ->
-      doc = "Alias defined in mix.exs"
+    Enum.reduce(aliases, {[], 0}, fn {alias_name, task}, {docs, max} ->
+      doc = alias_doc(task)
       task = "mix " <> alias_name
       {[{task, doc} | docs], max(byte_size(task), max)}
     end)
   end
 
+  defp alias_doc(task) do
+    "Alias for " <> format_alias_doc(task)
+  end
+
+  defp format_alias_doc(task) when is_list(task) do
+    task
+    |> Enum.map(&format_alias_task/1)
+    |> Enum.join(", ")
+  end
+
+  defp format_alias_doc(task), do: format_alias_doc([task])
+
+  defp format_alias_task(task) when is_binary(task), do: task
+
+  defp format_alias_task(task) when is_function(task) do
+    info = Function.info(task)
+    name = Atom.to_string(info[:name])
+
+    cond do
+      info[:type] == :remote -> inspect(task)
+      info[:type] == :local and String.contains?(name, "/") -> "a function"
```

Comment:
I couldn't find a better way to differentiate between

```
aliases: [
    foo: &foo/1,
    bar: fn _ -> bar() end
]
```

since both are `:local`